### PR TITLE
Added extra props and CSS handles for styling

### PR DESCRIPTION
**Pull request by razvanudream**
#### What is the purpose of this pull request?
The PR's purpose is to add extra CSS handles and props to deal with styling certain blocks, re-arranging them in a certain way (as per client's request), displaying extra info (with no impact on default implementations due to the fact that the new props have a default value)

#### What problem is this solving?
Helps provide a solution for a new VTEX client who requested a special design for the confirmation page

#### How should this be manually tested?

1. Link the changes on a workspace with the default order-placed template
2. Turn on some of the newly added props
3. See the magic happen :) 

#### Screenshots or example usage
image

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: messages/de.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
   "store/header.newpurchase.button": "Neue Bestellung starten",
   "store/header.print.button": "Drucken",
   "store/header.thanks": "Vielen Dank für den Kauf!",
+  "store/header.thanks.name": "Vielen Dank für den Kauf, {lastName} {firstName}!",
```
Comment (camilavcoutinho) [inline messages/de.json:14]: "Vielen Dank für Ihren Kauf, {lastName} {firstName}!",

File: messages/es.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
   "store/header.newpurchase.button": "Empezar nuevo orden",
   "store/header.print.button": "Imprimir",
   "store/header.thanks": "Gracias por la compra!",
+  "store/header.thanks.name": "Gracias por la compra, {lastName} {firstName}!",
```
Comment (camilavcoutinho) [inline messages/es.json:14]: "¡Gracias por la compra, {lastName} {firstName} !",

File: messages/es.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
   "store/header.newpurchase.button": "Empezar nuevo orden",
   "store/header.print.button": "Imprimir",
   "store/header.thanks": "Gracias por la compra!",
```
Comment (camilavcoutinho) [inline messages/es.json:13]: "¡Gracias por la compra!",

File: messages/es.json [hunk 4]
```diff
@@ -40,9 +41,11 @@
   "store/payments.id": "ID de pago: {id}",
   "store/payments.installments": "({installments}x)",
   "store/payments.transaction.id": "ID de transacción: {id}",
+  "store/payments.billing.title": "Dirección de envio",
```
Comment (camilavcoutinho) [inline messages/es.json:44]: "Dirección de facturación",

File: messages/it.json [hunk 5]
```diff
@@ -11,6 +11,7 @@
   "store/header.newpurchase.button": "Inizia un nuovo ordine",
   "store/header.print.button": "Stampa",
   "store/header.thanks": "Grazie per il tuo acquisto!",
```
Comment (camilavcoutinho) [inline messages/it.json:13]: "Grazie per l'acquisto!",

File: messages/it.json [hunk 6]
```diff
@@ -11,6 +11,7 @@
...
+  "store/header.thanks.name": "Grazie per il tuo acquisto, {firstName} {lastName}!",
```
Comment (camilavcoutinho) [inline messages/it.json:14]: "Grazie per l'acquisto, {firstName} {lastName}!",

File: messages/nl.json [hunk 7]
```diff
@@ -40,9 +41,11 @@
   "store/payments.id": "Betaal-ID: {id}",
   "store/payments.installments": "({installments}x)",
   "store/payments.transaction.id": "Transactie-ID: {id}",
+  "store/payments.billing.title": "Facturatie adres",
```
Comment (camilavcoutinho) [inline messages/nl.json:44]: "Factuuradres",

**Review by camilavcoutinho — CHANGES_REQUESTED** [review]

**Review by camilavcoutinho — APPROVED** [review]
For translations, LGTM!

File: CHANGELOG.md [hunk 8]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added - 2021-03-09
```
Comment (klzns) [inline CHANGELOG.md:10]:
```suggestion
### Added
```

File: docs/README.md [hunk 9]
```diff
@@ -519,12 +524,47 @@ Renders the customer information. Must be placed inside an [`op-order`](#op-orde
 | -------------------------------------------------- |
 | ![op-order-payment](https://user-images.githubusercontent.com/17712401/89225037-eec36f80-d5af-11ea-82af-267cb69c7d7b.png) |
 
+### `op-order-billing`
+
+Renders the billing details set in CustomData during checkout (must have the following fields: `name`, `phone_number`, `city`, `street` defined ). Must be placed inside an [`op-order`](#op-order) block.
```
Comment (klzns) [inline docs/README.md:529]: Custom data is not the same across stores, this should be a custom component instead.

File: react/ConfirmationTitle.tsx [hunk 10]
```diff
@@ -2,16 +2,34 @@ import React, { FC } from 'react'
...
+        <FormattedMessage
+          id="store/header.thanks.name"
+          values={{
+            firstName: profile.firstName,
+            lastName: profile.lastName,
+          }}
+        />
+      ) : (
+        <FormattedMessage id="store/header.thanks" />
+      )}
```
Comment (klzns) [inline react/ConfirmationTitle.tsx:32]: There's no need for the prop `displayName`, we can just provide the values and the user chooses in the translation message if they want to use it the `firstName` and `lastName`.

```suggestion
      <FormattedMessage
        id="store/header.thanks"
        values={{
          firstName: profile.firstName,
          lastName: profile.lastName,
        }}
      />
```

File: react/OrderBilling.tsx [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+interface Props {
+  displayTitle?: boolean
+}
+
+const OrderBilling: FC<Props> = ({ displayTitle = true }) => {
+  const { customData } = useOrder()
+  const handles = useCssHandles(CSS_HANDLES)
+
+  if (!customData || customData.customApps.length === 0) return null
```
Comment (klzns) [inline react/OrderBilling.tsx:24]: As the name says, CustomData varies from store to store. This component should be a custom component instead.

File: react/components/PickUpHeader.tsx [hunk 12]
```diff
@@ -19,7 +20,12 @@ const CSS_HANDLES = [
   'packageAdditionalInfo',
 ]
 
-const StorePickUpHeader: FC<Props> = ({ shippingData, index, numPackages }) => {
+const StorePickUpHeader: FC<Props> = ({
+  shippingData,
+  index,
+  numPackages,
+  displayTitle,
```
Comment (klzns) [inline react/components/PickUpHeader.tsx:27]:
```suggestion
  displayTitle = false,
```

File: react/components/DeliveryHeader.tsx [hunk 13]
```diff
@@ -17,13 +18,16 @@ const CSS_HANDLES = [
...
+  'packageAddressTitle',
+  'packageDeliveryTitle',
 ]
 
 const DeliveryHeader: FC<Props> = ({
   shippingData,
   index,
   numPackages,
   giftRegistry,
+  displayTitle,
```
Comment (klzns) [inline react/components/DeliveryHeader.tsx:30]:
```suggestion
  displayTitle = false,
```

File: store/interfaces.json [hunk 14]
```diff
@@ -48,6 +53,9 @@
   "op-order-number": {
     "component": "OrderNumber"
   },
+  "op-order-billing": {
+    "component": "OrderBilling"
+  },
```
Comment (klzns) [inline store/interfaces.json:58]: Remove this block because it doesn't suit every store

File: CHANGELOG.md [hunk 15]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+
+- Added `displayTitle` & `displayProducts` props to PickupPackages and DeliveryPackages
+- Added `displayName` prop to ConfirmationTitle
+- Added CSS Handles (`packageAddressTitle` & `packageDeliveryTitle`) to DeliveryHeader
+- Added new messages in all languages
+- Added `op-order-billing` component
```
Comment (klzns) [inline CHANGELOG.md:16]:
```suggestion
### Added

- `displayTitle` and `displayProducts` props to PickupPackages and DeliveryPackages
- Added CSS Handles (`packageAddressTitle` & `packageDeliveryTitle`) to DeliveryHeader
- Added new messages in all languages
```

**Review by klzns — CHANGES_REQUESTED** [review]
Thanks for the pull request @razvanudream! I left some things to change in it.

File: docs/README.md (hunk 9 above)
Comment (klzns) [reply docs/README.md:529]: Now anyone can create React components in apps:
https://developers.vtex.com/vtex-developer-docs/changelog/react-builder-in-vtex-io-now-in-open-beta

Comment (razvanudream) [reply docs/README.md]: Removed everything related billing from order-placed app and moved it to a custom app instead.

File: react/ConfirmationTitle.tsx (hunk 10 above)
Comment (razvanudream) [reply react/ConfirmationTitle.tsx]: Removed the prop and will handle this from the translation itself.

Comment (razvanudream) [reply react/ConfirmationTitle.tsx]: Also removed the extra message I added ("store/header.thanks.name")

File: react/OrderBilling.tsx (hunk 11 above)
Comment (razvanudream) [reply react/OrderBilling.tsx]: Removed everything related billing from order-placed app and moved it to a custom app instead.

File: react/components/PickUpHeader.tsx (hunk 12 above)
Comment (razvanudream) [reply react/components/PickUpHeader.tsx]: Added default value

File: react/components/DeliveryHeader.tsx (hunk 13 above)
Comment (razvanudream) [reply react/components/DeliveryHeader.tsx]: Added default value

File: store/interfaces.json (hunk 14 above)
Comment (razvanudream) [reply store/interfaces.json]: Removed everything related billing from order-placed app and moved it to a custom app instead.

File: CHANGELOG.md (hunk 15 above)
Comment (razvanudream) [reply CHANGELOG.md]: Updated changelog

**Review by clara-cruz — APPROVED** [review]

**Comment by klzns** [comment]
Thanks for the pull request! Since we can't commit on your branch we'll continue on https://github.com/vtex-apps/order-placed/pull/148

**Pull request closed without merging**